near/nep141-locker/src/lib.rs

```diff
@@ -716,16 +716,22 @@ impl Contract {
             .to_log_string(),
         );
 
-        ext_token::ext(token)
-            .with_static_gas(LOG_METADATA_GAS)
-            .with_attached_deposit(ONE_YOCTO)
-            .ft_transfer(fin_transfer.fee_recipient, U128(fee), None)
+        if fee > 0 {
+            PromiseOrValue::Promise(
+                ext_token::ext(token)
+                    .with_static_gas(LOG_METADATA_GAS)
+                    .with_attached_deposit(ONE_YOCTO)
+                    .ft_transfer(fin_transfer.fee_recipient, U128(fee), None),
+            )
+        } else {
+            PromiseOrValue::Value(fin_transfer.nonce)
```

Comment:
This could be a little bit confusing, because we are returning nonce just in one case. So let's return the unit `()` value instead

Reply:
We do the same thing in `fin_transfer_callback`, should I fix it as well for consistency?

Reply:
the `fin_transfer_callback` is a little bit different, the nonce is generated that indicate that a new transfer has been initialized.

Reply:
Fixed